### ISSUE TITLE
Replace chat @mentioning with a text-editable plugin

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#10804]
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)
### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)

Part of the username overhaul project underway is to make chat @mentioning prettier and more user-friendly. The majority of the implementation is accomplished on #10784, but given our use of a `textarea` component in chat, we weren't able to add styling for autocompleted @s within the text entry field itself. To make this possible, we'd need to replace our current autocomplete implementation, using a `contenteditable` div instead of a `textarea`. From there, we could employ an existing Vue plugin like https://github.com/fritx/vue-at to reimplement the rest of the functionality.

**Comment by jackcogdill** [comment]
Hey, wondering if I could work on this? I have some experience with `contenteditable` divs.

**Comment by paglias** [comment]
@jackcogdill thanks for the interest but I think this issue is a bit outdated as we actually addressed the contenteditable issue in this PR https://github.com/HabitRPG/habitica/pull/10872/files#diff-f9db085b45d481950984d26ae407e383R15 which is currently blocked for unrelated reasons

@SabreCat will be able to confirm

**Comment by SabreCat** [comment]
That is correct, @paglias, thanks for updating the labels! @jackcogdill, sorry for the misleading tag there--I hope you can find something else suited to your interests to work on!

**Comment by jackcogdill** [comment]
@SabreCat Thank you for following up. I was also wondering, is there a current feature request for adding keyboard support for the @mentions in the chat? It does not appear to be accessible currently.

**Comment by Alys** [comment]
@jackcogdill https://github.com/HabitRPG/habitica/issues/9291 "@Username selection with keyboard arrow keys in drop-down mentions list in chat"
If you comment there to say you'd like to work on it, we'll mark it as in progress for you. 

It's certainly an issue I'd love to see fixed!

**Comment by veeeeeee** [comment]
Even though the above mentioned PR has been merged, we're keep this issue on hold for a more in-depth look at how @mentioning works under the hood.
